main.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    result = []
+    my_each { |e| result.push e if yield e }
+    result
+  end
+
+  def my_all?
+    result = true
+    my_each do |e|
+      result = false unless yield e
+    end
+    result
+  end
+
+  def my_any?
+    result = false
+    my_each do |e|
+      if yield e
+        result = true
+        break
+      end
+    end
+    result
+  end
+
+  def my_none?
+    result = true
+    my_each do |e|
+      if yield e
+        result = false
+        break
+      end
+    end
+    result
+  end
+
```

Comment:
These methods should be able to accept arguments.
please compare your methods against the original like a`rr.my_all?(3) == arr.all?(3)`

**your method current yields this error LocalJumpError  no block given (yield)** You need to have